python/ray/_raylet.pyx

```diff
@@ -3444,6 +3454,11 @@ cdef class CoreWorker:
             CSchedulingStrategy c_scheduling_strategy
             c_vector[CObjectID] incremented_put_arg_ids
             c_string serialized_retry_exception_allowlist
+            CTaskID current_c_task_id
+            TaskID task_id_in_async_context = async_task_id.get()
+            # This task id is incorrect if async task is used.
```

Comment:
Can we always use contextvar for all types of actors not just async actors?

Reply:
I think it is possible, but if it s not a hard blocker, I prefer to properly fix it when we fix the whole thing in 2.8.

The reason is this requires us to properly "unset" task_id in various scenarios (e.g., task raises an exception). This is doable, but it requires more tests and stuff which increases the scope of the PR unnecessarily.